MuMuPiKPAT/test/sanjay/selector/psiPrimePiK_MC.C

```diff
@@ -5872,7 +5905,225 @@ void psiPrimePiK_MC::Terminate()
+              yy = hh_massGenDataFine->GetYaxis()->GetBinCenter(j+1);
+              if (!Dalitz_contour_host(xx, yy, kFALSE, 1)) hh_massGenDataFine->SetBinContent(i+1, j+1, 0);
+            }
+          TH1* hh_angGenDataFine = data_gen.createHistogram("genDatasetAngles", cosMuMu, Binning(nbins), YVar(phi, Binning(nbins))) ;
+          hh_massRecoPdfFine->Divide(hh_massGenDataFine);
+          hh_angRecoPdfFine->Divide(hh_angGenDataFine);
+
+          double intpdf0 = 0; double intpdf1 = 0;
+          int count0 = 0; int count1 = 0;
+          for (int i = 0; i < hh_massRecoPdfFine->GetNbinsX(); i++)
+            for (int j = 0; j < hh_massRecoPdfFine->GetNbinsY(); j++) {
+              intpdf0 += hh_massRecoPdfFine->GetBinContent(i+1, j+1);
+              if (hh_massRecoPdfFine->GetBinContent(i+1, j+1) != 0) count0++;
+            }
+          for (int i = 0; i < hh_angRecoPdfFine->GetNbinsX(); i++)
+            for (int j = 0; j < hh_angRecoPdfFine->GetNbinsY(); j++) {
+              intpdf1 += hh_angRecoPdfFine->GetBinContent(i+1, j+1);
+              if (hh_angRecoPdfFine->GetBinContent(i+1, j+1) != 0) count1++;
+            }
+          intpdf0 /= count0;
+          intpdf1 /= count1;
+          hh_massRecoPdfFine->Scale(1/intpdf0);
+          hh_angRecoPdfFine->Scale(1/intpdf1);
+
+          //Add exceeding bins to angles (missing)
+          hh_massRecoPdfFine->SetNameTitle("RelEff_psi2SPi_vs_KPi_hardCuts_1B0_rooKeys", "RelEff_psi2SPi_vs_KPi_hardCuts_1B0_rooKeys");
+          hh_massRecoPdfFine->Write();
+          hh_angRecoPdfFine->SetNameTitle("RelEff_planesAngle_vs_cos_psi2S_helicityAngle_hardCuts_1B0_rooKeys", "RelEff_planesAngle_vs_cos_psi2S_helicityAngle_hardCuts_1B0_rooKeys");
+          hh_angRecoPdfFine->Write();
+        }
+      }//if(MC)
+      else {
+        RooDataSet data_bkg("data_bkg", "data_bkg", RooArgSet(massKPi, massPsiPi, cosMuMu, phi));
+        dataTreeName = "hardcuts_variables_bkg";
+        Float_t obs1, obs2, obs3, obs4, obs5;
+        if ( (dataNTuple =(TNtupleD*)AlexisOut->Get(dataTreeName)) ) {
+          dataNTuple->SetBranchAddress("KPi_mass", &obs1);
+          dataNTuple->SetBranchAddress("PsiPi_mass", &obs2);
+          dataNTuple->SetBranchAddress("cosTheta_psi", &obs3);
+          dataNTuple->SetBranchAddress("phi", &obs4);
+          dataNTuple->SetBranchAddress("B0beauty", &obs5);
+          nEntries = dataNTuple->GetEntries();
+          for (Int_t i=0; i < nEntries; i++) {
+            dataNTuple->GetEntry(i);
+            massKPi = obs1;
+            massPsiPi = obs2;
+            cosMuMu = obs3;
+            phi = obs4;
+            if ( obs5 < 0.0) {
+              phi = -obs4;
+            }
+            data_bkg.add(RooArgSet(massKPi, massPsiPi, cosMuMu, phi));
+          }
+        } else {
+            cout << "ERROR in reading " << dataTreeName << "to compute background with RooKeys" << endl;
+            rooKeysError = kTRUE;
+        }
+        if (rooKeysError) {
```

Comment:
rooKeysError may be false also because of
https://github.com/lecriste/Z4430/pull/1/files#diff-2aef6870c472c96f6a5aec4107f14debR5924